### DISCUSSION TITLE
Duplicate check across users

### DISCUSSION THREAD

**Discussion by TheMarf** [simulot/immich-go#148] [Q&A] [answered]
Hello!  Let me start by saying this is what made it possible for me to be able to implement immich.  I am awful at CLI but the instructions and example commands make importing google takeout files painless.  I have a question about looking for duplicates across user keys.

Myself and my wife each have an acct with an API key that I used for import.  What I'm wondering is if I can run the below which omits the key.

`./immich-go -server=http://localip:2283  duplicate -yes`

I'm sure my wife and I have duplicates of each others photos since we text some back and forth etc.  Curious how this would work and if it would break anything?

**Comment by simulot** [comment]
The problem is that immich-go knows only photos of the the key owner. Therefore, the  duplicate search can't work accros users.

I surely can do something like following:
- accept more than one key
- get photos for all users
- run the deduplicaiton logic on the the full basis

Then what to do with cross user duplicates?
This problem could maybe addressed by the immich dev team. This is a very common use case. 
What do you have in mind?

**Reply by TheMarf** [reply, marked as answer]
Hey, thanks for the quick reply.  I do agree that this sounds like a very common use case.  I'm a brand new immich user so I don't have a full understanding of how it works at the library level.  Seems like hardlinking would be the best case and it'd probably need to be addressed by immich.  I did find [this post](https://github.com/immich-app/immich/discussions/4542) which is similar.
